LagFreeScreenshots/LagFreeScreenshotsMod.cs

```diff
@@ -0,0 +1,526 @@
+        private const string SettingCustomResolutionY = "CustomResolutionY";
+
+        private static MelonPreferences_Entry<bool> ourEnabled;
+        private static MelonPreferences_Entry<PresetScreenshotSizes> ourResolution;
+        private static MelonPreferences_Entry<string> ourFormat;
+        private static MelonPreferences_Entry<int> ourJpegPercent;
+        private static MelonPreferences_Entry<int> ourCustomResolutionX;
+        private static MelonPreferences_Entry<int> ourCustomResolutionY;
+        private static MelonPreferences_Entry<bool> ourAutorotation;
+        private static MelonPreferences_Entry<bool> ourMetadata;
+        private static MelonPreferences_Entry<int> ourRecommendedMaxFb;
+
+        private static Thread ourMainThread;
+
+        public override void OnApplicationStart()
+        {
+            var category = MelonPreferences.CreateCategory(SettingsCategory, "Lag Free Screenshots");
+            ourEnabled = category.CreateEntry(SettingEnableMod, true, "Enabled");
+            ourResolution = category.CreateEntry( SettingScreenshotResolution, PresetScreenshotSizes.Default, "Screenshot resolution override");
+            ourFormat = category.CreateEntry( SettingScreenshotFormat, "png", "Screenshot format");
+            ourJpegPercent = category.CreateEntry(SettingJpegPercent, 95, "JPEG quality (0-100)");
+            ourAutorotation = category.CreateEntry(SettingAutorotation, true, "Rotate picture to match camera");
+            ourMetadata = category.CreateEntry(SettingMetadata, false, "Save metadata in picture");
+            ourRecommendedMaxFb = category.CreateEntry(SettingRecommendedMaximumFb, 1024, "Try to keep framebuffer below (MB) by reducing MSAA");
+            ourCustomResolutionX = category.CreateEntry(SettingCustomResolutionX, 1920, "Custom screenshot resolution (X)");
+            ourCustomResolutionY = category.CreateEntry(SettingCustomResolutionY, 1080, "Custom screenshot resolution (Y)");
+            
+            HarmonyInstance.Patch(
+                typeof(PortableCamera).GetMethod(nameof(PortableCamera.Capture),  RefFlags.Instance | RefFlags.NonPublic),
+                new HarmonyMethod(AccessTools.Method(typeof(LagFreeScreenshotsMod), nameof(OnCapture))));
+        }
+
+        // FIXME: Those 2 were imported from UIExpansionKit and should be refactored somewhere else?
+        public override void OnUpdate()
+        {
+            TaskUtilities.ourMainThreadQueue.Flush();
+        }
+
+        public override void OnGUI()
+        {
+            TaskUtilities.ourFrameEndQueue.Flush();
+        }
+
+        private static ScreenshotRotation GetPictureAutorotation(Camera camera)
+        {
+            var pitch = Vector3.Angle(camera.transform.forward, new Vector3(0, 1, 0));
+            if (pitch < 45 || pitch > 135) return ScreenshotRotation.NoRotation; //Pointing up/down, rotation doesn't matter
+
+            var rot = camera.transform.localEulerAngles.z;
```

Comment:
```suggestion
            var rot = CVRCamController.Instance.cvrCamera.transform.localEulerAngles.z;
```

This _should_ fix the auto rotation bug